h2o-core/src/main/java/water/util/Log.java

```diff
@@ -339,31 +360,17 @@ private static void setLog4jProperties(String logDir, java.util.Properties p) th
 
     if (log4jConfigurationProvided) {
       PropertyConfigurator.configure(log4jConfiguration);
-    }
-    else {
+    } else {
       // Create some default properties on the fly if we aren't using a provided configuration.
       // H2O creates the log setup itself on the fly in code.
       java.util.Properties p = new java.util.Properties();
+      if (H2O.ARGS.log_dir != null) {
+        LOG_DIR = new File(H2O.ARGS.log_dir).getAbsolutePath();
+      } else {
+        LOG_DIR = defaultLogDir().getAbsolutePath();
+      }
```

Comment:
The exception thrown by `defaultLogDir` is already handled inside that method. If we are not able to create temp log directory after several attempts, we throw `RuntimeException`. The code above it does not throw any exception we need to check for so I made the try-catch block smaller.